internal/querynodev2/segments/segment_loader.go

```diff
@@ -75,6 +71,24 @@ type Loader interface {
 	LoadIndex(ctx context.Context, segment *LocalSegment, info *querypb.SegmentLoadInfo) error
 }
 
+type LoadResource struct {
+	MemorySize uint64
+	DiskSize   uint64
+	WorkNum    int
+}
+
+func (r *LoadResource) Add(resource LoadResource) {
```

Comment:
LoadResource or *LoadResource 🤔？

Reply:
Here it need to modify the struct, so receiver is a pointer. The struct is simple so we pass it by value to optimize for GC